base/linalg/uniformscaling.jl

```diff
@@ -31,6 +31,7 @@ eltype(::Type{UniformScaling{T}}) where {T} = T
 ndims(J::UniformScaling) = 2
 getindex(J::UniformScaling, i::Integer,j::Integer) = ifelse(i==j,J.λ,zero(J.λ))
 
+show(io::IO, J::UniformScaling{T}) where {T<:Complex} = print(io, "$(typeof(J))\n($(J.λ))*I")
```

Comment:
Aren't we already dealing with strings? My idea was to add the parentheses if something like `ismatch(r"[0-9]+ ?\+|\- ?[0-9]+", "$(J.λ)")`